### PR TITLE
Update engine

### PR DESCRIPTION
**Pull request by abarth**
Turns out there were more clients of the old paragraph API than I expected.
This patch migrates them to the new API.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: examples/layers/raw/text.dart [hunk 1]
```diff
@@ -72,11 +72,10 @@ void main() {
...
   paragraph = builder.build(new ui.ParagraphStyle())
-    // Next, we supply a maximum width that the text is permitted to occupy.
-    ..maxWidth = 180.0
-    // ... and we ask the paragraph to the visual position of each its glyphs as
-    // well as its overall size, subject to its sizing constraints.
-    ..layout();
+    // Next, we supply a width that the text is permitted to occupy and we ask
+    // the paragraph to the visual position of each its glyphs as well as its
+    // overall size, subject to its sizing constraints.
+    ..layout(new ui.ParagraphConstraints(width: 180.0));
```
Comment (Hixie) [inline examples/layers/raw/text.dart]: const?

**Comment by Hixie** [comment]
LGTM

File: examples/layers/raw/text.dart (hunk 1 above)
Comment (abarth) [inline examples/layers/raw/text.dart]: It's got an assert in its constructor.

**Pull request merged**